ml_metadata/tools/mlmd_bench/benchmark.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef ML_METADATA_TOOLS_MLMD_BENCH_BENCHMARK_H
+#define ML_METADATA_TOOLS_MLMD_BENCH_BENCHMARK_H
+
+#include <vector>
+
+#include "ml_metadata/metadata_store/types.h"
+#include "ml_metadata/tools/mlmd_bench/proto/mlmd_bench.pb.h"
+#include "ml_metadata/tools/mlmd_bench/workload.h"
+
+namespace ml_metadata {
+
+// Contains a list of workloads to be executed by ThreadRunner.
+// The executable workloads are generated according to `mlmd_bench_config`.
+class Benchmark {
+ public:
+  Benchmark(const MLMDBenchConfig& mlmd_bench_config);
+  ~Benchmark();
+
+  // Takes `workload_config` and generates corresponding executable workloads.
+  void CreateWorkload(const WorkloadConfig& workload_config);
```

Comment:
should this be a internal anonymous namespace method?
also consider to use a template CreateWorkload(const T& config, ...), where the T is the one of those WorkloadConfigs, use a list of specialized CreateWorkload<FillTypeWorkloadConfig) to implement them separately. wdyt?

Reply:
Yes, my bad. It should be a internal anonymous namespace method. However, as for the template method, will it be too much difference compared to the current method? Since every workload has different constructor to pass in which means we have to create many branches as well? wdyt?